zallet/src/components/json_rpc/methods/view_transaction.rs

```diff
@@ -257,6 +366,95 @@ pub(crate) fn call(wallet: &DbConnection, txid_str: &str) -> Response {
+        // Transparent inputs
+        for (input, idx) in bundle.vin.iter().zip(0u16..) {
+            let txid_prev = input.prevout().txid().to_string();
+
+            // TODO: Migrate to a hopefully much nicer Rust API once we migrate to the new Zaino ChainIndex trait.
+            let (address, value) = match chain.get_raw_transaction(txid_prev.clone(), Some(1)).await
+            {
+                Ok(GetRawTransaction::Object(tx)) => {
+                    let output = tx
+                        .outputs()
+                        .get(usize::from(idx))
+                        .expect("Zaino should have rejected this earlier");
+                    (
+                        transparent::address::Script::from(output.script_pub_key().hex())
+                            .address()
+                            .map(|addr| addr.encode(wallet.params())),
+                        Zatoshis::from_nonnegative_i64(output.value_zat())
+                            .expect("Zaino should have rejected this earlier"),
+                    )
+                }
+                Ok(_) => unreachable!(),
+                Err(_) => todo!(),
+            };
+
+            transparent_input_values.insert(input.prevout(), value);
+
+            spends.push(Spend {
+                pool: POOL_TRANSPARENT,
+                t_in: Some(idx),
+                spend: None,
+                action: None,
+                txid_prev,
+                t_out_prev: Some(
+                    input
+                        .prevout()
+                        .n()
+                        .try_into()
+                        .expect("should always be small enough"),
+                ),
+                output_prev: None,
+                action_prev: None,
+                address,
+                value: value_from_zatoshis(value),
+                value_zat: value.into_u64(),
+            });
+        }
+
+        // Transparent outputs
+        for (output, idx) in bundle.vout.iter().zip(0..) {
+            let (address, outgoing, wallet_internal) = match output.recipient_address() {
+                None => (None, true, false),
+                Some(address) => {
+                    let wallet_scope =
+                        wallet
+                            .get_account_ids()
+                            .unwrap()
+                            .into_iter()
+                            .find_map(|account| {
+                                match wallet.get_transparent_address_metadata(account, &address) {
+                                    Ok(Some(metadata)) => Some(metadata.scope()),
+                                    _ => None,
+                                }
+                            });
+
+                    (
+                        Some(address.encode(wallet.params())),
+                        wallet_scope.is_none(),
+                        matches!(wallet_scope, Some(TransparentKeyScope::INTERNAL)),
```

Comment:
Why does this not also count ephemeral addresses as internal?

Reply:
Because they are not internal; they are a different scope, and they are addresses that are used to interact with external parties (in a specific controlled way) and thus are "exposed externally".

Also, per `zcashd`, `wallet_internal` is documented as `"true" if this is a change output`, and ephemeral addresses are not used for change outputs.